direct-query/src/main/java/org/opensearch/sql/directquery/transport/model/ExecuteDirectQueryActionRequest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.directquery.transport.model;
+
+import java.io.IOException;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.sql.directquery.rest.model.ExecuteDirectQueryRequest;
+
+/*
+ * @opensearch.experimental
+ */
+public class ExecuteDirectQueryActionRequest extends ActionRequest {
+  private final ExecuteDirectQueryRequest directQueryRequest;
+
+  public ExecuteDirectQueryActionRequest(ExecuteDirectQueryRequest directQueryRequest) {
+    super();
+    this.directQueryRequest = directQueryRequest;
+  }
+
+  public ExecuteDirectQueryActionRequest(StreamInput in) throws IOException {
+    // In a real implementation, deserialize the request
+    // This is just a placeholder since we don't have the full serialization code
+    this.directQueryRequest = new ExecuteDirectQueryRequest();
+  }
+
+  @Override
+  public void writeTo(StreamOutput out) throws IOException {
+    super.writeTo(out);
+    // Add serialization logic if needed
+  }
```

Comment:
i had the question since beginning: are these actionable TODO items, or are we ok to just leave them? I believe this is for transport layer communications, which we don't have a use case, so "real implementation" would mean when this request is being used at transport laayer?